packages/map-template/src/components/RouteInstructions/RouteInstructions.jsx

```diff
@@ -18,22 +24,65 @@ function RouteInstructions({ steps, onNextStep, onPreviousStep, originLocation }
     /** Referencing the previous step of each active step */
     const [previous, setPrevious] = useState();
 
-    const [activeStep, setActiveStep] = useState(0);
+    const [activeStep, setActiveStep] = useRecoilState(activeStepState);
+    const [totalSteps, setTotalSteps] = useState();
+
+    const [lastStepZoom, setLastStepZoom] = useState();
+    const [lastStepCenter, setLastStepCenter] = useState();
+
+    const directions = useRecoilValue(directionsResponseState);
+
+    const mapsIndoorsInstance = useRecoilValue(mapsIndoorsInstanceState);
 
-    /*
-     * Make sure to reset active step whenever the steps change.
+    /**
+     * Clone the last step in the directions in order to create a destination step.
+     * Assign the specific travel mode to the destination step and push the destination step at the end of the steps array.
+     *
      */
     useEffect(() => {
-        setActiveStep(0);
+        const lastStep = steps[steps.length - 1];
+        const destinationStep = { ...lastStep }
+        destinationStep.travel_mode = 'DESTINATION'
+        steps.push(destinationStep);
+        setTotalSteps(steps);
     }, [steps]);
 
+    /**
+     * Get the zoom and the center of the last step and the destination step in the directions.
+     *
+     */
+    useEffect(() => {
+        if (activeStep === totalSteps?.length - 1) {
+            const lastStepZoom = mapsIndoorsInstance.getZoom();
+            const lastStepCenter = mapsIndoorsInstance.getMapView().getCenter();
+            setLastStepZoom(lastStepZoom);
```

Comment:
Will this be wrong if the user has panned and/or zoomed the map?

Reply:
Hmm, I don't see why it would be wrong, I have tested it and it seems to be working fine.

Reply:
See here:

https://github.com/MapsPeople/web-ui/assets/50314165/02282361-30f0-4a87-b680-e4d2bc156bc0

Maybe we should calculate the bounds of the route and use that instead.

Reply:
That's right, good point. So your idea would be that we calculate the bounds of the route instead of having the lastStepZoom and the lastStepCenter?

Reply:
Something like that.

But we can also try to see if the SDK's `setStepIndex` method can be used: https://app.mapsindoors.com/mapsindoors/js/sdk/latest/docs/mapsindoors.directions.DirectionsRenderer.html#setStepIndex

Will it refit the bounds when calling with an index that is the current index?

Reply:
I tried to use the `setStepIndex` but it doesn't seem to work. I might be doing something wrong - looking into it.

Reply:
How can we calculate the bound of the route however?

Reply:
Hmmm yeah, I don't think we can in an easy manner, since we also need to take into account if the user has rotated and/or tilted the map.

Another idea could be to find a way to know when the map has panned to the last step. I think the mapView instance will emit an `idle` event whenever the map stops moving. Maybe we can use that.

Reply:
Added event listener when map instance is idle ✅